tracer/src/Datadog.Trace/IAST/EvidenceConverter.cs

```diff
@@ -0,0 +1,240 @@
+    }
+
+    private void WriteRedactedValuePart(JsonWriter writer)
+    {
+        writer.WriteStartObject();
+        writer.WritePropertyName("redacted");
+        writer.WriteValue(true);
+        writer.WriteEndObject();
+    }
+
+    private void WriteRedactedValuePart(JsonWriter writer, Range range)
+    {
+        writer.WriteStartObject();
+        writer.WritePropertyName("redacted");
+        writer.WriteValue(true);
+        if (range.Source != null)
+        {
+            writer.WritePropertyName("source");
+            writer.WriteValue(range.Source.GetInternalId());
+        }
+
+        writer.WriteEndObject();
+    }
+
+    private Range? Poll(LinkedList<Range> list)
+    {
+        if (list.First != null)
+        {
+            var res = list.First.Value;
+            list.RemoveFirst();
+            return res;
+        }
+
+        return null;
+    }
+
+    private void ToRedactedJson(JsonWriter writer, string value, Range[]? taintedRanges, Range[]? sensitiveRanges)
+    {
+        writer.WriteStartArray();
+        int start = 0;
+        LinkedList<Range> tainted = taintedRanges != null ? new LinkedList<Range>(taintedRanges) : new LinkedList<Range>();
+        LinkedList<Range> sensitive = sensitiveRanges != null ? new LinkedList<Range>(sensitiveRanges) : new LinkedList<Range>();
+
+        Range? nextTainted = Poll(tainted);
+        Range? nextSensitive = Poll(sensitive);
+        for (int i = 0; i < value.Length; i++)
+        {
+            if (nextTainted != null && nextTainted.Value.Start == i)
+            {
+                WriteValuePart(writer, value.Substring(start, i - start));
+                // clean up contained sensitive ranges
+                while (nextSensitive != null && nextTainted.Value.Contains(nextSensitive.Value))
+                {
+                    nextTainted.Value.Source?.MarkAsRedacted();
+                    nextSensitive = Poll(sensitive);
+                }
+
+                if (nextSensitive != null && nextSensitive.Value.Intersects(nextTainted.Value))
+                {
+                    nextTainted.Value.Source?.MarkAsRedacted();
+                    var rest = nextSensitive.Value.Remove(nextTainted.Value);
+                    nextSensitive = rest.Count > 0 ? rest[0] : null;
+                }
+
+                if (nextTainted.Value.Source != null && nextTainted.Value.Source.IsRedacted())
+                {
+                    WriteRedactedValuePart(writer, nextTainted.Value);
+                }
+                else
+                {
+                    WriteValuePart(writer, Substring(value, nextTainted.Value), nextTainted);
+                }
+
+                start = i + nextTainted.Value.Length;
+                i = start - 1;
+                nextTainted = Poll(tainted);
+            }
+            else if (nextSensitive != null && nextSensitive.Value.Start == i)
+            {
+                WriteValuePart(writer, value.Substring(start, i - start));
```

Comment:
This line is changed in `Source.cs` it there's an extra `+ 1`  there:

```
                WriteValuePart(writer, value.Substring(start, i - start + 1));
```

Are both correct?

Reply:
The SourceConverter code is old. Already removed it.